stan/math/prim/err/check_cholesky_factor.hpp

```diff
@@ -25,14 +25,14 @@ namespace math {
  *   if there are 0 columns, or if any element in matrix is NaN
  */
 template <typename T_y>
-inline void check_cholesky_factor(
-    const char* function, const char* name,
-    const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y) {
+inline void check_cholesky_factor(const char* function, const char* name,
+                                  const T_y& y) {
```

Comment:
Q/Optional: There's no other specializations of this but would we want T_y to fail during sfinae instead of after it selected the function? I'm not sure which one is nicer

Reply:
Which error condition is this about?

Reply:
If T_y was not an eigen matrix

Reply:
I think it does not matter. For consistency with other functions I think it is fine as it is.